opensrp-reveal/src/main/java/org/smartregister/reveal/widget/GeoWidgetFactory.java

```diff
@@ -50,10 +56,28 @@
 
     private static final String ZOOM_LEVEL = "zoom_level";
 
+    private static final String MAX_ZOOM_LEVEL = "v_zoom_max";
+
     private RevealMapView mapView;
 
     private JsonApi jsonApi;
 
+    public static ValidationStatus validate(JsonFormFragmentView formFragmentView, RevealMapView mapView) {
+        if (!Utils.isEmptyCollection(mapView.getValidators())) {
+            for (METValidator validator : mapView.getValidators()) {
+                MapboxMap mapboxMap = mapView.getMapboxMap();
+                double zoom = mapView.getMapboxMap().getCameraPosition().zoom;
+                Log.d(TAG, "Current Zoom level: " + zoom);
+                if (mapboxMap != null && !validator.isValid(String.valueOf(zoom), false)) {
```

Comment:
Does this assume that only one validator will be added? 

Only the validator that confirms a valid zoom level is specified?

Reply:
What happens if one wants to add another validator on the map view?

Reply:
In native form one error is returned per widget. This would allow the user to address an issue at a time